app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java

```diff
@@ -125,129 +122,69 @@ private static void checkIfApiReady(Context context) {
-                StrictMode.setVmPolicy(cacheVmPolicy);
-                return null;
-            }
-            // although we should check for weakApplicationContext.get() every time before we use it,
-            // but since it's an application context so we should be fine here.
-            // but since it's an application context so we should be fine here.
-            final Context applicationContext = weakApplicationContext.get();
-
             // this is only for testing. So we can simulate slow network..etc
             final BlockingEnablerCallback callback = FirebaseHelper.enablerCallback;
             if (callback != null) {
                 callback.runDelayOnExecution();
             }
 
-            setDeveloperModeEnabled(AppConstants.isFirebaseBuild());
-
-            // make sure we are in the changing state
-            changing = true;
-
             // this methods is blocking.
-            updateInstanceId(applicationContext, enable);
-
-            enableCrashlytics(applicationContext, enable);
-            enableAnalytics(applicationContext, enable);
-            enableCloudMessaging(applicationContext, RocketMessagingService.class.getName(), enable);
-            enableRemoteConfig(applicationContext, enable, () -> {
-                ThreadUtils.postToBackgroundThread(() -> {
-                    final String pref = applicationContext.getString(R.string.pref_s_news);
-                    final String source = getRcString(applicationContext, pref);
-                    final Settings settings = Settings.getInstance(applicationContext);
-                    final boolean canOverride = settings.canOverride(PREF_INT_NEWS_PRIORITY, Settings.PRIORITY_FIREBASE);
-                    Log.d(NewsSourceManager.TAG, "Remote Config fetched");
-                    if (!TextUtils.isEmpty(source) && (canOverride || TextUtils.isEmpty(settings.getNewsSource()))) {
-                        Log.d(NewsSourceManager.TAG, "Remote Config is used:" + source);
-                        settings.setPriority(PREF_INT_NEWS_PRIORITY, Settings.PRIORITY_FIREBASE);
-                        settings.setNewsSource(source);
-                        NewsSourceManager.getInstance().setNewsSource(source);
-                    }
-                });
-            });
-            // now firebase has completed state changing,
-            changing = false;
-            // we'll check if the cached state is the same as our current one. If not, issue
-            // a state change again.
-            if (pending != null && pending != enable) {
-                enableFirebase(applicationContext, pending);
-            } else {
-                // after now, there'll be now pending state.
-                pending = null;
-            }
-
-            StrictMode.setThreadPolicy(cachedThreadPolicy);
-            StrictMode.setVmPolicy(cacheVmPolicy);
+            Companion.getInstance().deleteInstanceId();
```

Comment:
this call shouldn't happen in the main thread.
we only call this when the users disable "Send Usage Data"
Any call to Remote Config fetching, or push notification.getGcmToken() after this call, will fetch a new Instance ID automatically.